server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java

```diff
@@ -236,6 +246,140 @@ public void testCancel() throws Exception {
         });
     }
 
+    /**
+     * Makes sure that lookup fields are resolved on the local cluster
```

Comment:
Yes, let's discuss this. I chose to resolve lookup fields in only the local cluster to have a consistent result with or without ccs_minimize_roundtrips.

Reply:
We discussed this offline and agreed it'd be more natural for the lookup field to be resolved within the same cluster. Related note: I think it would be totally fine if we didn't want to support `ccs_minimize_roundtrips: false` at first. It is not the default and some new features like vector tiles do not support it (https://www.elastic.co/guide/en/elasticsearch/reference/7.17/modules-cross-cluster-search.html#ccs-network-delays).

Reply:
I don't have all the context, but it may be good to mention that async search does not support minimizing roundtrips, and that Kibana relies on async search nowadays. It seems like in general we should discuss what to do about minimizing roundtrips.

Reply:
This is an important point, then we should indeed support `ccs_minimize_roundtrips: false`. +1 to discuss separately what to do about minimizing roundtrips, it seems it's causing complexity.

Reply:
Sorry for not being clear. The issue here is not related to ccs_minimize_roundtrips. It's about the user experience (i.e., providing the least surprising behavior for the feature). Also, we don't know if a search request fetches lookup runtime fields defined in the index mappings on the coordinating node where we decide to enable or disable the ccs_minimize_roundtrips.